### ISSUE TITLE
Vite asset misses some files when processed

### ISSUE THREAD

**Issue by Rizky92** [laravel/vite-plugin#132]
- Laravel Version: 9.26.1
- PHP Version: 8.1.6
- Database Driver & Version: N/A

Others:  
- NodeJS v16
- OS Windows 10

I followed the [documentation about processing static assets](https://laravel.com/docs/9.x/vite#blade-processing-static-assets) to process my static asset files. However, some files were missed by vite.  
  
My files contains a lot of repeated filenames and identical structure, with only difference is the origin name of the directory, like example:  
```
resources/
  static/
    subdomain-a/
      logo.png
      logo-alternate.png
      ...
    subdomain-b/
      logo.png
      logo-alternate.png
      ...
    subdomain-c/
      logo.png
      logo-alternate.png
      ...
...
```

Repo: https://github.com/Rizky92/bug-report

**Comment by driesvints** [comment]
Transferred this to the vite-plugin repo.

**Comment by timacdonald** [comment]
@Rizky92 whoa. that was a rabbit hole I just went down on this one!

It seems that files that are 100% identical are causing this issue.

I've created a PR to Vite in an attempt to fix this, although I'm not 100% sure if this is the "correct" fix from a Vite perspective (build tools are hard).

Attempted fix: https://github.com/vitejs/vite/pull/9928

For now you may have to create a "shared" directory and put any shared resources across clients in there (i.e. any files that are globed that _exactly_ identical).

Hopefully we can get a resolution for this via Vite.

**Comment by Rizky92** [comment]
Thank you very much @timacdonald.
I'll try to work around that one. I've checked that apparently the only file with 100% identical was the manifest file.

Once again, thank you.

**Comment by timacdonald** [comment]
Will close this as this isn't a bug in the plugin. We can follow along upstream and see if we can get anything sorted.

**Comment by timacdonald** [comment]
Looks like the PR is being considered for Vite `3.1`

**Comment by timacdonald** [comment]
Just a heads up that the upstream fix for this has now been tagged in vite `3.2.0`.
